analysis/tests/src/expected/TypeContextCompletion.res.txt

```diff
@@ -0,0 +1,12 @@
+Complete src/TypeContextCompletion.res 10:44
+posCursor:[10:44] posNoWhite:[10:43] Found expr:[10:11->18:1]
+Pexp_apply ...[10:11->10:30] (~someVariant10:32->10:43=...[13:0->17:3])
+Completable: CnamedArg(Value[someVariantToString], "", [someVariant])
```

Comment:
So this is where it gets a bit hairy. Not sure why it decides it's a named arg completion given that it's in the expression assigned to the label. It could be because that expression is exactly where parser recovery happened and some heuristic is needed because of the uncertainty that parser recovery brings.
Something orthogonal, but need to figure it out in order to understand this example.

Anyhow the `CnamedArg` completion finds the function name, and the labels already visited: `[someVariant]` so they are not repeated. This is not going to find anything as it's the only label.

Reply:
OK looking at the code, I think we're hitting this case:
```ml
    | [] ->
      if posAfterFunExpr <= posBeforeCursor && posBeforeCursor < endPos then
        Some (CnamedArg (contextPath, "", allNames))
```

So we went through the entire application and did not find anything interesting. So we'll just default to named arg completion.
This can anyway be overridden by visiting sub-expressions later. The innermost AST expression where something interesting is found wins. Because the innermost is the most precise hit of the cursor.

To see why this case is needed one can just turn it off and see what tests fail.
Looks like this is one such example: ```let x = Lib.foo(~```